### ISSUE TITLE
Unable to create "Data_Group" using "externalFilePath": Failed! exit_code (22)

### ISSUE THREAD

**Issue by devildog-tech** [F5Networks/f5-appsvcs-extension#586]
### Environment
 * Application Services Version: 3.35.0
 * BIG-IP Version: 16.1.2.1

### Summary
Utilizing "class": "Data_Group" and "externalFilePath" option fails to deploy with the following error. The intent is to create an external data-group with source located on gitlab. I have validated the url and token to authenticate multiple times utilizing curl from the same F5. I also utilized the same url's and sources for other class's such as i-rules with no issues. There is no other indication in any of the logs with debug and trace turned on that I can see to indicate the reason for the failure. I also have tried string type datagroup and ip type with the same result.

The impact is currently we can not migrate to AS3 without the ability to create data-groups from our source of truth on gitlab.

```
"code": 422,
"message": "declaration failed",
"response": "Failed! exit_code (22).\n",
"host": "localhost",
"tenant": "testApp",
"runTime": 3112
```

### Steps To Reproduce

Data-group formats I have tried. Obviously changing the keyDataType depending on the data.

`"TEST" := "TESTVALUE",`

`TEST := "TESTVALUE",`

`TEST,`

`10.2.1.1,`

`10.2.1.1 := "TESTVALUE"`

Declaration to reproduce.

```
{
    "$schema": "https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/latest/as3-schema.json",
    "class": "AS3",
    "action": "deploy",
    "persist": false,
    "id": "testApp",
    "logLevel": "debug",
    "trace": true,
    "traceResponse": true,
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.0.0",
        "testApp": {
            "class": "Tenant",
            "testApp": {
                "class": "Application",
                "template": "generic",
                "testApp": {
                    "class": "Data_Group",
                    "storageType": "external",
                    "keyDataType": "string",
                    "separator": ":=",
                    "externalFilePath": {
                        "url": "https://removed/api/v4/projects/879/repository/files/roles%2Fremoved%2Ffiles%2Fltm%2Fdatagroup%2FtestApp.txt/raw?ref=main",
                        "authentication": {
                            "method": "bearer-token",
                            "token": "removed"
                        }
                    }
                }
            }
        }
    }
}
```

```
"results": [
        {
            "code": 422,
            "message": "declaration failed",
            "response": "Failed! exit_code (22).\n",
            "host": "localhost",
            "tenant": "testApp",
            "runTime": 3112
        }
    ],
```

### Expected Behavior
The expectation is to be able to include my hundreds of data-groups throughout 1000's of tenants keeping the truth of source on GitLab.

### Actual Behavior
I'm currently to get any data-groups to create successfully with external file path using token with very limited error output.

```
"results": [
        {
            "code": 422,
            "message": "declaration failed",
            "response": "Failed! exit_code (22).\n",
            "host": "localhost",
            "tenant": "testApp",
            "runTime": 3112
        }
    ],
```

**Comment by dstokesf5** [comment]
@devildog-tech Thank you for your feedback. The `externalFilePath` property is mapped to the `source-path` property in tmsh. Unfortunately, when tmsh fails to fetch that data, you see this 22 exit code error message. We are improving that experience by catching bad URLs sooner in the declaration handling. Unfortunately, those improvements will not help here because tmsh does not support token auth with the `source-path` property. We will need to update AS3 to handle the URL fetching ourselves when token auth is used. I have added AUTOTOOL-3111 to our internal product backlog to address this issue.

**Comment by devildog-tech** [comment]
Is it possible to prioritize this issue? This bug is causing the inability for me to create external datagroups at all. In order to point to file I have to authenticate. It’s also causing extremely large declarations to skirt around. Because I’m having to run a jinja2 template that takes the datagroup files and writes them to record key value pairs. Some of my datagroups are huge, which are not ideal to be internal datagroups, and also not ideal to write into the declaration. Thank you.

**Comment by dstokesf5** [comment]
A developer is currently working on this issue.

**Comment by dstokesf5** [comment]
This has been resolved in the 3.38.0 release of AS3.
